File: filter_log.rb
OK_WORDS = [
  "warning: ISO C90 forbids mixed declarations and code",
]

def match(line)
  OK_WORDS.each do |word|
    return true if line.include? word
  end
  false
end

if __FILE__ == $0
  input = ARGV[0]

  arr = []
  `cat #{input} | grep warning`.split("\n").each do |line|
    if match(line)
      next
    end
    arr << line
  end
  $stdout << arr.join("\n")
end

File: Driver/filter_log.py
import sys 

OK_WORDS = [
	"In function",		
	"ISO C90 forbids mixed declarations",
]

def do_filter(lines):
	_lines = []
	for line in lines:
		b = True
		for w in OK_WORDS:
			b &= not (w in line)
		if(b):
			_lines.append(line)
	return _lines	

if __name__ == '__main__':
	filename = sys.argv[1]
	lines = open(filename, 'r').read().split("\n")
	s = '\n'.join(do_filter(lines)).strip()
	print(s)
